openquake/hazardlib/geo/surface/simple_fault.py

```diff
@@ -203,11 +201,9 @@ def get_fault_patch_vertices(cls, fault_trace, upper_seismogenic_depth,
         t_lon = []
         t_lat = []
         t_dep = []
-
         for point in fault_trace.points:
-            top_edge_point = point.point_at(hdist_top, 0, azimuth)
+            top_edge_point = point
```

Comment:
The top_edge_point  should be the points in fault_trace directly, because here the input fault_trace is actually already the top edge of the rupture area.(the name of fault_trace is a bit miss leading).